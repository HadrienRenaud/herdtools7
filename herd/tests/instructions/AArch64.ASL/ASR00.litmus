AArch64 ASR00
{
uint64_t 0:X8=0xff000000000000ff;
uint64_t 0:X9;
}

  P0                ;
ORR X9,XZR,X8,ASR 8 ;
ASRV X9,X9,XZR      ;
ASR  X9,X9,#8       ;

forall
0:X9=0xffffff0000000000