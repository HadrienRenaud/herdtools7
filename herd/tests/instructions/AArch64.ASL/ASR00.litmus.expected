Test ASR00 Required
States 1
0:X9=18446742974197923840;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X9=-1099511627776)
Observation ASR00 Always 1 0
Hash=53510b8fe12f32a7ab06716773da0a35

